src/accelerate/utils/dataclasses.py

```diff
@@ -861,7 +872,7 @@ def save_model(self, accelerator, model, output_dir, model_index=0):
             with FSDP.state_dict_type(model, self.state_dict_type, self.state_dict_config):
                 state_dict = model.state_dict()
         else:
-            FSDP.set_state_dict_type(model, self.state_dict_type, self.state_dict_config)
+            FSDP.set_state_dict_type(model, self.state_dict_type, self.state_dict_config, self.optim_state_dict_config)
```

Comment:
is `self.optim_state_dict_config` required for saving model. If yes, I assume the same param should be passed in `load_model`, right?

Reply:
From looking at the PyTorch code, the optimizer config shouldn't be needed in either load or save model.  I'll remove it from here.